tests/unit/FileApiTest.php

```diff
@@ -106,47 +106,52 @@ public function testUploadFile()
-                        $this->authProvider->getTokenType(),
-                        $this->authProvider->getAccessToken(),
-                    ]),
-                ],
-                'exceptions' => false,
-                'multipart' => [
-                    [
-                        'name' => 'authorize',
-                        'contents' => 0,
-                    ],
-                    [
-                        'name' => 'smartling.client_lib_id',
-                        'contents' => '{"client":"smartling-api-sdk-php","version":"4.0.0"}',
-                    ],
-                    [
-                        'name' => 'localeIdsToAuthorize[]',
-                        'contents' => 'es',
-                    ],
-                    [
-                        'name' => 'smartling.namespace',
-                        'contents' => 'namespace_for_deduplication',
-                    ],
-                    [
-                        'name' => 'file',
-                        'contents' => $this->streamPlaceholder,
-                    ],
-                    [
-                        'name' => 'fileUri',
-                        'contents' => 'test.xml',
+            ->willReturnCallback(function(string $method, string $uri, array $options) {
+                $this->assertEquals('post', $method);
+                $this->assertEquals(FileApi::ENDPOINT_URL . '/' . $this->projectId . '/file', $uri);
+                $this->assertEquals([
+                    'headers' => [
+                        'Accept' => 'application/json',
+                        'Authorization' => \vsprintf('%s %s', [
+                            $this->authProvider->getTokenType(),
+                            $this->authProvider->getAccessToken(),
+                        ]),
                     ],
-                    [
-                        'name' => 'fileType',
-                        'contents' => 'xml',
+                    'exceptions' => false,
+                    'multipart' => [
+                        [
+                            'name' => 'authorize',
+                            'contents' => 0,
+                        ],
+                        [
+                            'name' => 'smartling.client_lib_id',
+                            'contents' => '{"client":"smartling-api-sdk-php","version":"4.0.2"}',
+                        ],
+                        [
+                            'name' => 'localeIdsToAuthorize[]',
+                            'contents' => 'es',
+                        ],
+                        [
+                            'name' => 'smartling.namespace',
+                            'contents' => 'namespace_for_deduplication',
+                        ],
+                        [
+                            'name' => 'file',
+                            'contents' => $this->streamPlaceholder,
+                            'filename' => 'unknown'
```

Comment:
Once again, this change makes it somewhat easier. Previous phpunit error message was something like
```
Parameter 2 for invocation GuzzleHttp\Client::request('post', 'https://api.smartling.com/fil...Id/uri', Array (...)): Psr\Http\Message\ResponseInterface does not match expected value.
Failed asserting that two arrays are equal.
```
Now it's displaying the failing assertion

Reply:
Maybe change `unknown` -> `filename_not_set`

Reply:
Done